### PR TITLE
Skip injections for subframes of trusted pages

### PR DESCRIPTION
**Pull request by philipp-classen**
Heuristic to detect navigations of subframes of trusted pages and skip them.

refs https://github.com/ghostery/ghostery-extension/issues/1690
refs https://github.com/ghostery/broken-page-reports/issues/697

### CODE REVIEW

File: extension-manifest-v3/src/background/adblocker.js [hunk 1]
```diff
@@ -326,7 +339,10 @@ if (__PLATFORM__ === 'safari') {
   });
 } else {
   chrome.webNavigation.onCommitted.addListener(async (details) => {
-    injectScriptlets(details.tabId, details.url);
+    const isMainFrame = details.parentFrameId === -1;
+    if (isMainFrame || !(await isPausedTab(details.tabId))) {
```
Comment (smalluban) [inline extension-manifest-v3/src/background/adblocker.js:343]: The `details` object contains the `url`. Why do you need to use `chrome.tabs` API? We can simply call `isPaused(parse(details.url).hostname)`

**Review by smalluban — CHANGES_REQUESTED** [review]

File: extension-manifest-v3/src/background/adblocker.js (hunk 1 above)
Comment (smalluban) [reply extension-manifest-v3/src/background/adblocker.js:343]: Reference: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webNavigation/onCommitted#url

Comment (smalluban) [inline extension-manifest-v3/src/background/adblocker.js:343]: I have an idea to use `tabStats` here synchroniusly. As the mainframe must call this event before subframes, we can find stats for the `details.tabId` and have the `stats.hostname` here.

Comment (smalluban) [reply extension-manifest-v3/src/background/adblocker.js:343]:
```suggestion
    if (isMainFrame || !(await isPaused(tabStats.get(details.tabId)?.hostname) {
```

**Review by smalluban — CHANGES_REQUESTED** [review]

File: extension-manifest-v3/src/background/adblocker.js (hunk 1 above)
Comment (philipp-classen) [reply extension-manifest-v3/src/background/adblocker.js:343]: This is a follow-up that tries out the idea: https://github.com/ghostery/ghostery-extension/pull/1692

**Comment by philipp-classen** [comment]
Closed in favor of https://github.com/ghostery/ghostery-extension/pull/1692

**Pull request closed without merging**